remix.init/index.js

```diff
@@ -0,0 +1,136 @@
+];
+
+const filesToCopy = [["README.md"], ["netlify.toml"]];
+
+const filesToModify = ["app/entry.server.tsx", "app/root.tsx"];
+
+async function modifyFilesForEdge(files, rootDirectory) {
+  const filePaths = files.map((file) => join(rootDirectory, file));
+  const contents = await Promise.all(
+    filePaths.map((path) => fs.readFile(path, "utf8"))
+  );
+
+  await Promise.all(
+    contents.map((content, index) => {
+      const newContent = content.replace(
+        /@remix-run\/node/g,
+        "@netlify/remix-runtime"
+      );
+      return fs.writeFile(filePaths[index], newContent);
+    })
+  );
+}
+
+async function copyTemplateFiles(files, rootDirectory) {
+  for (const [file, target] of files) {
+    await fs.copyFile(
+      join(rootDirectory, "remix.init", file),
+      join(rootDirectory, target || file)
+    );
+  }
+}
+
+async function updatePackageJsonForEdge(directory) {
+  const packageJson = await PackageJson.load(directory);
+  const {
+    dependencies: {
+      "@remix-run/netlify": _netlify,
+      "@remix-run/node": _node,
+      ...dependencies
+    },
+    scripts,
+    ...restOfPackageJson
+  } = packageJson.content;
+
+  packageJson.update({
+    // dev script is the same as the start script for Netlify Edge, "cross-env NODE_ENV=production netlify dev"
+    scripts: {
+      ...scripts,
+      dev: scripts["start"],
+    },
+    ...restOfPackageJson,
+    dependencies: {
+      ...dependencies,
+      "@netlify/edge-functions": "^2.0.0",
+      "@netlify/remix-edge-adapter": "1.0.0",
+    },
+  });
+
+  await packageJson.save();
+}
+
+async function removeNonTemplateFiles({ rootDirectory, folders }) {
+  try {
+    await Promise.allSettled(
+      folders.map((folder) =>
+        fs.rm(join(rootDirectory, folder), { recursive: true, force: true })
+      )
+    );
+  } catch (e) {
+    console.log(
+      `Unable to remove folders ${folders.join(
+        ", "
+      )}. You can remove them manually.`
+    );
+  }
+}
+
+async function main({ rootDirectory }) {
+  await removeNonTemplateFiles({
+    rootDirectory,
+    folders: foldersToExclude,
+  });
+
+  if (!(await shouldUseEdge())) {
+    copyTemplateFiles(filesToCopy, rootDirectory);
+
+    return;
+  }
+
+  await Promise.all([
+    fs.mkdir(join(rootDirectory, ".vscode")),
+    copyTemplateFiles(edgeFilesToCopy, rootDirectory),
+  ]);
+
+  await Promise.all([
+    modifyFilesForEdge(filesToModify, rootDirectory),
+    updatePackageJsonForEdge(rootDirectory),
+  ]);
+}
+
+async function shouldUseEdge() {
+  const { edge } = await inquirer.prompt([
+    {
+      name: "edge",
+      type: "list",
+      message: "Run your Remix site with:",
+      choices: [
+        {
+          name: "Netlify Functions - Choose this for stable support for production sites",
+          value: false,
+        },
+        {
+          name: "Netlify Edge Functions (beta) - Try this for improved performance on non-critical sites",
```

Comment:
I’m waiting to hear back about wording for this. @stephmarie17 notnsure what was decided for this.

Reply:
Checking back on that Slack thread, this wording is fine especially since it includes the beta label, if anything @joeyaiello  made the following suggestion:

```suggestion
          name: "Netlify Edge Functions (beta) - Try this for improved performance on less critical sites",
```

Reply:
Thanks @stephmarie17!